### PR TITLE
accel: add internal folder

### PR DESCRIPTION
**Pull request by siserafin**
Adding accel team folder and internal packages

### CODE REVIEW

File: tests/accel/internal/accelconfig/config.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	*config.GeneralConfig
+}
+
+// NewAccelConfig returns instance of AccelConfig type.
+func NewAccelConfig() *AccelConfig {
+	log.Print("Creating new AccelConfig")
+
+	var accelConfig AccelConfig
+	accelConfig.GeneralConfig = config.NewConfig()
+
```
Comment (kononovn) [inline tests/accel/internal/accelconfig/config.go:29]: I think you need to add here:


```
	err := envconfig.Process("eco_accel_upgrade_", accelConfig)
	if err != nil {
		log.Printf("failed to instantiate AccelConfig: %v", err)

		return nil
	}
```

File: tests/accel/internal/accelconfig/config.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-gotests/tests/internal/config"
+)
+
+// AccelConfig contains environment information related to ocp upgrade tests.
+type AccelConfig struct {
+	PullSecret           string `envconfig:"ECO_ACCEL_PULL_SECRET"`
+	Registry             string `envconfig:"ECO_ACCEL_REGISTRY"`
+	UpgradeTargetVersion string `envconfig:"ECO_ACCEL_UPGRADE_TARGET_IMAGE"`
+	HubKubeConfig        string `envconfig:"ECO_ACCEL_HUB_KUBECONFIG"`
```
Comment (kononovn) [inline tests/accel/internal/accelconfig/config.go:16]: HubKubeConfig is always default KUBECONFIG variable. If you need secondary connection to Spoke please add SpokeKubeConfig instead.

File: tests/accel/internal/accelconfig/config.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	HubClusterName       string `envconfig:"ECO_ACCEL_HUB_CLUSTER_NAME"`
+	HubMinorVersion      string `envconfig:"ECO_ACCEL_HUB_MINOR_VERSION"`
+	HubAPIClient         *clients.Settings
```
Comment (kononovn) [inline tests/accel/internal/accelconfig/config.go:19]: Please change it to SpokeAPIClient

File: tests/accel/internal/accelinittools/accelinittools.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+var (
+	// HubAPIClient provides API access to hub cluster.
+	HubAPIClient *clients.Settings
+	// AccelConfig provides access to configuration parameters.
+	AccelConfig *accelconfig.AccelConfig
+)
+
+func init() {
+	HubAPIClient = inittools.APIClient
```
Comment (kononovn) [inline tests/accel/internal/accelinittools/accelinittools.go:17]: Please add here:
SpokeAPIClient = AccelConfig.SpokeAPIClient

**Review by kononovn — CHANGES_REQUESTED** [review]
Please fix few comments

File: tests/accel/internal/accelconfig/config.go (hunk 3 above)
Comment (siserafin) [reply tests/accel/internal/accelconfig/config.go]: Changed, thanks

File: tests/accel/internal/accelinittools/accelinittools.go (hunk 4 above)
Comment (siserafin) [reply tests/accel/internal/accelinittools/accelinittools.go:19]: Done, thanks

**Review by kononovn — APPROVED** [review]
/lgtm

**Review by trewest — APPROVED** [review]
/lgtm

File: tests/accel/internal/accelconfig/config.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	if accelConfig.SpokeKubeConfig != "" {
+		glog.V(90).Infof("Creating spoke api client from %s", accelConfig.SpokeKubeConfig)
+
+		if accelConfig.SpokeAPIClient = clients.New(
+			accelConfig.SpokeKubeConfig); accelConfig.SpokeAPIClient == nil {
+			glog.V(90).Infof("failed to load provided spoke kubeconfig")
+		}
+	}
```
Comment (kononovn) [inline tests/accel/internal/accelconfig/config.go:45]: Add:

```
else {
		accelConfig.SpokeAPIClient = nil
	}
```

**Pull request merged**